runtime/vm/resolvesupport.cpp

```diff
@@ -1847,6 +1847,106 @@ resolveMethodHandleRef(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpInde
 
 }
 
+j9object_t
+resolveConstantDynamic(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpIndex, UDATA resolveFlags)
+{
+	J9RAMConstantDynamicRef *ramCPEntry = (J9RAMConstantDynamicRef*)ramCP + cpIndex;
+	j9object_t value = ramCPEntry->value;
+	J9JavaVM *vm = vmThread->javaVM;
+	bool resolved = false;
+
+retry:
+	/* Check if already resolved */
+	if (NULL != value) {
+		resolved = true;
+	} else if (NULL != ramCPEntry->exception) {
+		J9Class *throwable = J9VMJAVALANGTHROWABLE_OR_NULL(vm);
+		if (isSameOrSuperClassOf(throwable, J9VM_J9CLASS_FROM_HEAPCLASS(vmThread, ramCPEntry->exception))) {
+			/* Resolved with exception */
+			VM_VMHelpers::setExceptionPending(vmThread, ramCPEntry->exception);
+			resolved = true;
+		} else if (ramCPEntry->exception == vm->voidReflectClass->classObject) {
+			/* Resolved with null reference */
+			resolved = true;
+		} else if (ramCPEntry->exception != vmThread->threadObject) {
+			/* Another thread is currently trying to resolve the entry */
+			omrthread_monitor_enter(vm->constantDynamicMutex);
+			omrthread_monitor_wait(vm->constantDynamicMutex);
+			omrthread_monitor_exit(vm->constantDynamicMutex);
```

Comment:
Wait should not be used bare like this - correct use is to wait for a condition to become true.